src/ripple/app/misc/impl/TxQ.cpp

```diff
@@ -1393,6 +1393,41 @@ TxQ::getMetrics(OpenView const& view) const
     return result;
 }
 
+std::tuple<XRPAmount, std::uint32_t, std::uint32_t>
+TxQ::getTxRequiredFeeAndSeq(OpenView const& view,
+    std::shared_ptr<STTx const> const& tx) const
+{
+    auto const account = (*tx)[sfAccount];
+
+    std::lock_guard lock(mutex_);
+
+    auto const snapshot = feeMetrics_.getSnapshot();
+    auto const baseFee = calculateBaseFee(view, *tx);
+    auto fee = FeeMetrics::scaleFeeLevel(snapshot, view);
+
+    auto const sle = view.read(keylet::account(account));
+
+    std::uint32_t accountSeq;
+    if (sle)
+        accountSeq = (*sle)[sfSequence];
+    else
+        accountSeq = 0;
+
+    auto availableSeq = accountSeq;
+
+    if (auto accIter {byAccount_.find(account)}; accIter != byAccount_.end())
+    {
+        auto& txQAcct = accIter->second;
+        for (auto iter : txQAcct.transactions)
```

Comment:
The variable is named `iter`, but it is a `std::pair` not a type of iterator. Also, `iter` is taken by value, this makes an unneeded copy (the first element of the pair is cheap to copy, the second is not).

However, I think the clearest change is to use structured binding to just get the first element. Something like:
```
        for (auto const& [seq, _] : txQAcct.transactions)
        {
            if (seq >= availableSeq)
                availableSeq = seq + 1;
        }
```

Reply:
Done